### ISSUE TITLE
Optimize Mesh of an Indoor Scene

### ISSUE THREAD

**Issue by spr-serena** [NVlabs/nvdiffrec#159]
Hi,

thank you for your great work! However, I'm wondering if this project can be used to optimize meshes that reconstructs indoor scenes? I'm currently optimizing my mesh based on a set of RGB images and camera poses, but it doesn't work well.

My goal: optimize RGB color and position for each vertex, ignore PBR materials and environment light.

Since I'm reconstructing an indoor scene, so I apply alpha = 255 throughout the images for masking.

Here's my json file. I read through other discussions after the training, so I know the texture_res might be too high, but the texture besides the noise still seems to be incorrect.
```
{
    "base_mesh": "scannet.obj",
    "ref_mesh": "data/nerf_synthetic/scannet",
    "random_textures": true,
    "iter": 5000,
    "save_interval": 100,
    "texture_res": [ 2048, 2048 ],
    "train_res": [468, 624],
    "batch": 1,
    "learning_rate": [0.03, 0.01],
    "ks_min" : [0, 0.25, 0.0],
    "laplace_scale" : 3000,
    "display": [{"latlong" : true}, {"bsdf" : "kd"}, {"bsdf" : "ks"}, {"bsdf" : "normal"}],
    "background" : "white",
    "out_dir": "scannet"
}
```

I'm using images from Scannet like this:

![0](https://github.com/NVlabs/nvdiffrec/assets/81208129/747773cc-8fd4-4078-b5ab-e1e5955e3934)

Here are my results:

![img_mesh_pass_000050](https://github.com/NVlabs/nvdiffrec/assets/81208129/9d515c0e-32dd-457f-a296-209c3a7fa089)
<img width="459" alt="kd" src="https://github.com/NVlabs/nvdiffrec/assets/81208129/4bc15423-35d7-4479-a7a4-ffab8703faf4">
<img width="446" alt="output_mesh" src="https://github.com/NVlabs/nvdiffrec/assets/81208129/a6774ec3-a447-401f-8897-0bf2e7033781">

Any suggestions would be helpful. Thank you so much!

**Comment by JHnvidia** [comment]
Hi Serena,

Unfortunately one of the limitations of the method is that you need a clear foreground (object) / background separation with the mask. Density based methods, (nerf/neus/neuralangelo) perform better on full scenes. If you can supervise with a reasonably accurate depth estimate, it might be possible to make training converge, but I don't expect it to work without mask.
